sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2Strategy.scala

```diff
@@ -245,6 +246,28 @@ object DataSourceV2Strategy extends Strategy with PredicateHelper {
       }.toArray
       DeleteFromTableExec(r.table.asDeletable, filters) :: Nil
 
+    case UpdateTable(r: DataSourceV2Relation, attrs, values, condition) =>
+      val nested = attrs.asInstanceOf[Seq[Any]].filterNot(_.isInstanceOf[AttributeReference])
```

Comment:
why do we need the `.asInstanceOf[Seq[Any]]`?

Reply:
If not, it will regard `attrs` to be of type `Seq[Attribute]`, and `filterNot` will treat each element as `Attribute`. 
As you metioned in https://github.com/apache/spark/pull/25626#discussion_r320095937, a nested filed will not be resolved to `Attribute`, but `GetStructField` or something other, so if we change the type of `attrs` to `Seq[Expression]`, we can eliminate `.asInstanceOf[Seq[Any]]` here.